app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -288,6 +290,18 @@ def create_vm(args = {})
       vm
     end
 
+    def id_name_substitution(argument_list, argument)
+      argument_list.detect { |a| a.name == argument }.try(:id) || argument
+    end
+
+    def get_verified_argument(argument_list, argument)
```

Comment:
From the name it's not clear what this does.
Maybe something like `get_ovirt_id`?

Reply:
Yes, this could be a good name. I wanted to also include the fact that the function verifies that the given argument name is a valid name. I will change it